shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/controller/VipClientService.java

```diff
@@ -0,0 +1,223 @@
+package org.shanoir.ng.vip.controller;
+
+import jakarta.annotation.PostConstruct;
+import org.keycloak.representations.AccessTokenResponse;
+import org.shanoir.ng.shared.exception.ErrorModel;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.shanoir.ng.shared.exception.SecurityException;
+import org.shanoir.ng.shared.security.KeycloakServiceAccountUtils;
+import org.shanoir.ng.utils.KeycloakUtil;
+import org.shanoir.ng.vip.dto.VipExecutionDTO;
+import org.shanoir.ng.vip.resulthandler.ResultHandlerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientException;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+@Service
+public class VipClientService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(VipClientService.class);
+
+    @Value("${vip.uri}/executions/")
+    private String vipExecutionUrl;
+
+    @Value("${vip.uri}/pipelines/")
+    private String vipPipelineUrl;
+
+    @Autowired
+    private KeycloakServiceAccountUtils keycloakServiceAccountUtils;
+
+    private WebClient webClient;
+
+    @PostConstruct
+    public void init() {
+        this.webClient = WebClient.create();
```

Comment:
Hi Youenn, I hope you are doing fine and thank you for your PR!
I would propose, if possible to add .baseUrl("https://vip"); after the
create and work with get(uri), either executions or pipelines.
In the idea, as you always call the same server, that the opened
http connection (url + port -> socket) is re-used again and again.
To assure this, it might be worth going by baseUrl, just an idea.
Maybe WebClient is already so smart internally. With kind regards,
Michael